http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/websocket/WebSocketTestCases.java

```diff
@@ -0,0 +1,153 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.wso2.carbon.transport.http.netty.websocket;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.messaging.exceptions.ServerConnectorException;
+import org.wso2.carbon.transport.http.netty.config.TransportsConfiguration;
+import org.wso2.carbon.transport.http.netty.config.YAMLTransportConfigurationBuilder;
+import org.wso2.carbon.transport.http.netty.listener.HTTPServerConnector;
+import org.wso2.carbon.transport.http.netty.util.TestUtil;
+import org.wso2.carbon.transport.http.netty.util.clients.websocket.WebSocketClient;
+import org.wso2.carbon.transport.http.netty.util.clients.websocket.WebSocketTestConstants;
+
+import java.net.URISyntaxException;
+import java.nio.ByteBuffer;
+import java.util.List;
+
+import javax.net.ssl.SSLException;
+
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertTrue;
+
+/**
+ * Test class for WebSocket Upgrade
+ */
+public class WebSocketTestCases {
+
+    Logger logger = LoggerFactory.getLogger(WebSocketTestCases.class);
+    private List<HTTPServerConnector> serverConnectors;
+    private WebSocketClient primaryClient = new WebSocketClient();
+    private WebSocketClient secondaryClient = new WebSocketClient();
```

Comment:
Primary client is the one who is checking the connections of the Server. When secondary server is connecting to the endpoint message will be sent to the primary client indicating the state of the secondary client.